# 02 - Markdown and Formatting

Jupyter notebooks are not just for code — you can mix Markdown and code to create clear, professional-looking reports.

This notebook introduces:

- Writing Markdown cells (headings, lists, links)
- Adding inline LaTeX math
- Embedding images
- Why formatting matters in data science communication


## 1. Headings and Text Styles

### Example

# Heading 1
## Heading 2
### Heading 3

**Bold text**, *italic text*, `inline code`


## My Favorite Data Tools

- **Python**
- **Pandas**
- **Jupyter Notebook**


## 2. Lists

### Example

- Item 1
- Item 2
  - Sub-item 2a
  - Sub-item 2b

1. First
2. Second
3. Third


### My Top 3 Programming Languages

1. Python
2. JavaScript
3. C++


## 3. Links and Images

### Links

[OpenAI](https://openai.com)

### Images

![MSU Logo](https://upload.wikimedia.org/wikipedia/en/thumb/8/82/Midwestern_State_University_seal.svg/512px-Midwestern_State_University_seal.svg.png)

### Your Turn

Here is a link to a favorite website and an embedded image:

[Wikipedia](https://www.wikipedia.org)

![Notebook Image](https://upload.wikimedia.org/wikipedia/commons/3/38/Jupyter_logo.svg)


## 4. LaTeX Math

You can render math using LaTeX syntax inside `$ ... $` for inline or `$$ ... $$` for block math.

### Example

Inline: $y = mx + b$

Block:

$$
E = mc^2
$$

### Your Turn

Quadratic formula:

$$
x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}
$$


## 5. Mixing Code and Markdown

In [3]:
# Example: create data in Python
numbers = [1, 2, 3, 4, 5]
squares = [n**2 for n in numbers]
squares

[1, 4, 9, 16, 25]

Now we explain results in Markdown:

- The list `numbers` contained 1–5.
- The list `squares` contains each number squared.
- Mixing narrative + results is why notebooks are so powerful.


### Your Turn

Create a list of 3 numbers, calculate their cubes in code, then explain in Markdown.


In [4]:
nums = [2, 4, 6]
cubes = [n**3 for n in nums]
cubes

[8, 64, 216]

The list `nums` contains the values 2, 4, and 6.

The list `cubes` contains each of those values raised to the third power:
- 2³ = 8
- 4³ = 64
- 6³ = 216

This shows how code can generate results and Markdown can explain what they mean.


## Summary

- Use headings and formatting to organize.
- Lists and links make content readable.
- LaTeX lets you add professional math.
- Combine code + Markdown for clear data storytelling.
